core/app/services/spree/cart/add_item.rb

```diff
@@ -12,7 +12,7 @@ def call(order:, variant:, quantity: nil, options: {})
 
       private
 
-      def add_to_line_item(order:, variant:, quantity: nil, options: {})
+      def add_to_line_item(order:, variant:, quantity: nil, public_metadata: nil, private_metadata: nil, options: {})
         options ||= {}
```

Comment:
```suggestion
```

Reply:
Error occurs when params[:options] is nil. In that case it overrides the default {} and causes error https://github.com/spree/spree/blob/9c125ee29d65d35e9ff897fc6db8a653239507b0/core/app/services/spree/cart/add_item.rb#L24 here.